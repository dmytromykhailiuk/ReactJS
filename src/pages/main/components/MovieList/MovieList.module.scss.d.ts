// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'movie-list': string;
}
export const cssExports: CssExports;
export default cssExports;
