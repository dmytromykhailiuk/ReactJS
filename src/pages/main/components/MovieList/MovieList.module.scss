@import "scss/mixins";

.movie-list {
  list-style: none;

  & > * {
    display: inline-block;
    
    @include small-screen {
      margin-right: 4vw;

      &:nth-child(2n) {
        margin-right: 0;
      }
    }
  
    @include large-screen {
      &:nth-child(n) {
        margin-right: 4vw;
      }

      &:nth-child(3n) {
        margin-right: 0;
      }
    }
  }
}
