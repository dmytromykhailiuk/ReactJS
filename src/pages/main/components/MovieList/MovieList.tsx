import React from "react";
import { Movie } from "models";
import { MovieItem } from "../MovieItem";
import classes from "./MovieList.module.scss";

interface MovieListProps {
  movies: Movie[];
}

const MovieList: React.FC<MovieListProps> = ({ movies }) => {
  return (
    <ul className={classes["movie-list"]}>
      {
        movies.map(movie => <li key={movie.key}><MovieItem movie={movie}/></li>)
      }
    </ul>
  );
}

export default MovieList;
