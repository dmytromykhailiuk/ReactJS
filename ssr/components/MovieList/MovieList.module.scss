@import "../../styles/mixins";

.movie-list {
  list-style: none;

  &>* {
    display: inline-block;

    @include small-screen {
      margin-right: 4vw;

      &:nth-child(2n) {
        margin-right: 0;
      }
    }

    @include large-screen {
      &:nth-child(n) {
        margin-right: 3vw;
      }

      &:nth-child(3n) {
        margin-right: 0;
      }
    }

    @include extra-large-screen {
      &:nth-child(n) {
        margin-right: 2vw;
      }

      &:nth-child(4n) {
        margin-right: 0;
      }
    }
  }
}