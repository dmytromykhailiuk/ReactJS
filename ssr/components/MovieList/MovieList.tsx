import React from "react";
import { Movie } from "../../models";
import { MovieItem } from "../MovieItem";
import classes from "./MovieList.module.scss";

interface MovieListProps {
  movies: Movie[];
  onEditMovie: (movie: Movie) => void;
  onDeleteMovie: (movie: Movie) => void;
}

const MovieList: React.FC<MovieListProps> = ({ movies, onEditMovie, onDeleteMovie, }) => {
  return (
    <ul className={classes["movie-list"]}>
      {
        movies.map((movie) => (
          <li key={movie.id}>
            <MovieItem
              movie={movie}
              onEditMovie={onEditMovie}
              onDeleteMovie={onDeleteMovie}
            />
          </li>
        ))
      }
    </ul>
  );
};

export default MovieList;
