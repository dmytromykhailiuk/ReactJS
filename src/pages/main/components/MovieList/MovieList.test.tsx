import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import MovieList from './MovieList';

describe('MovieList', () => {
  it('should match snepshot', () => {
    const wrapper = shallow(
      <MovieList movies={[{ id: 13211 }] as any[]} onEditMovie={() => {}} onDeleteMovie={() => {}} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
