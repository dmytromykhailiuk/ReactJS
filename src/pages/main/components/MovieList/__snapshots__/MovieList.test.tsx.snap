// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should match snepshot 1`] = `
<ul
  className="movie-list"
>
  <li
    key="13211"
  >
    <Memo()
      movie={
        Object {
          "id": 13211,
        }
      }
      onDeleteMovie={[Function]}
      onEditMovie={[Function]}
    />
  </li>
</ul>
`;
